### ISSUE TITLE
Major Kong issue with backend routing in Kong 3.7.1 in Service resources vs 2.8.x.

### ISSUE THREAD

**Issue by jeremyjpj0916** [Kong/kong#13393]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

3.7.1

### Current Behavior

On Kong 2.8.x we did not observe this. In Kong 3.7.1 we do:

The service resource we have like so:
```
{
  "id": "76d80c70-b0eb-4a38-bf55-d5d06c5349ad",
  "write_timeout": 9000,
  "protocol": "https",
  "tls_verify": null,
  "tls_verify_depth": null,
  "read_timeout": 9000,
  "host": "myapi.company.com",
  "tags": null,
  "created_at": 1713416244,
  "updated_at": 1713416244,
  "path": "/api/test/moretest/v2.0/",
  "client_certificate": null,
  "enabled": true,
  "retries": 0,
  "name": "boss.boss.popproductv2.stage",
  "ca_certificates": null,
  "port": 443,
  "connect_timeout": 2000,
  "extras": {}
}
```

Kong 2.8.x would not drop the trailing / when proxying. New Kong 3.7.1 testing it routes to the backside without the / so it tries: ```/api/test/moretest/v2.0``` when it goes to the backend and it returns 404 now because that trailing slash is relevant to API responses. Would like fixes asap if possible.

**Comment by jeremyjpj0916** [comment]
router_flavor = traditional_compatible we do not override this field so it stays defaulted to this in our runtimes.

**Comment by jeremyjpj0916** [comment]
Some more insights for yall after further testing since last night was a mad dash to fix whatever I could however I could:

This test essentially does not pass in real life with functionality:

https://github.com/Kong/kong/blob/master/spec/fixtures/router_path_handling_tests.lua#L91 

  "path_handling": "v1" on route calling /route path, on 2.8.x series will respect the service resource path trailing slash like ```/service/``` in proxy calls


  "path_handling": "v1" on route calling /route, on 3.7.1 series will NOT respect the service trailing slash like ```/service/``` in service resource path , actually sends traffic to ```/service``` in behavior and drops the trailing /. 
  
  This is breaking behavior for some APIs.

**Comment by chronolaw** [comment]
path_handling 'v1' is deprecated since Kong gateway 3.0, the only available value is 'v0', I think that we should find a warning message in the log.

**Comment by chronolaw** [comment]
See change logs of 3.0 and this PR (https://github.com/Kong/kong/pull/9290).

**Comment by jeremyjpj0916** [comment]
@chronolaw deprecated is very different from breaking behavior. An application should honor existing functionality even when deprecated.

**Comment by jeremyjpj0916** [comment]
That is a potential option I had considered but there are performance loss with that mode right? And traditional is going away is my understanding hence why we continued with 3.7.x default provided value.

And does Kong have a plan to migrate v1 path handlings to v0 when v1 will be moved from deprecated to deleted? Kinda odd Kong is doing reverse versioning there where v0 is preferred but v1 is the deprecated, usually +1 version iterations are the next iteration of a feat. vs considered legacy. And in the future I suppose the field should go away completely if there was only one in the future, shouldn't be configurable maybe in 4.x . I still think this is a pretty glaring issue in between the 2.x to 3.x versions for customers that use Kong. Some of the new routing behaviors of Kong don't seem to make common sense anymore for this issue and the other one I raised.

**Comment by gchristidis** [comment]
We are looking at Kong LTS upgrade (2.8 to 3.4) for our systems and have been going though the [Kong Gateway 2.8 to 3.4 LTS upgrade - Kong Gateway | Kong Docs](https://developer.konghq.com/gateway/upgrade/lts-upgrade/) document preparing everything only to find on our first test of config we came across this issue and now have to go back and check everything.

Kong does log the following when creating/updating a route with v1 path_handling
```
025-07-22 13:51:49 2025/07/22 05:51:49 [warn] 1294#0: *1196 [kong] routes.lua:117 path_handling='v1' is deprecated and will not work under 'traditional_compatible' router_flavor, please use path_handling='v0' instead (deprecated after 3.0), client: 172.18.0.1, server: kong_admin, request: "PUT /routes/e98e8947-dd02-4e30-b8fa-7519292b098f HTTP/1.1", host: "localhost:8001"
```

The [Routes - Kong Gateway | Kong Docs](https://developer.konghq.com/gateway/entities/route/) documentation shows both v0 and v1 are supported in path_handling with a warning box indicating that v1 is not supported in the expressions router, which in [Expressions router - Kong Gateway | Kong Docs](https://developer.konghq.com/gateway/routing/expressions/) is used when router_flavor = expressions. It does not mention anything about v1 not compatible with router_flavor = traditional_compat.  There is also no mention of the fact that v1 is in fact ignored and defaults to v0

There is also no mention of path_handling as a breaking change in the [Kong Gateway 2.8 to 3.4 LTS upgrade - Kong Gateway | Kong Docs](https://developer.konghq.com/gateway/upgrade/lts-upgrade/) or the [Kong Gateway breaking changes - Kong Gateway | Kong Docs](https://developer.konghq.com/gateway/breaking-changes/#3-0-x-breaking-changes).

This is almost as big a breaking change as prefixing REs as all routes automatically change to v0 even when v1 is specified which in most cases, depending on route path, route strip_path and incoming request path will break most configured routes  It is not mentioned in any documents.

**Comment by jeremyjpj0916** [comment]
Yeah it was def a goof on Kong's end w how they handled the transition. Luckily only broke a few of our proxies and we managed workarounds.
